File: hw/cuichan/u3/hw37StrategyPattern/SortApp/src/main/java/ec/edu/espe/sortapp/controller/SortingContext.java
package ec.edu.espe.sortapp.controller;

import ec.edu.espe.sortapp.model.BubbleSort;
import ec.edu.espe.sortapp.model.InsertionSort;
import ec.edu.espe.sortapp.model.QuickSort;
import ec.edu.espe.sortapp.model.SortingStrategy;

/**
 *
 * @author David Cuichan
 */
public class SortingContext {
    private SortingStrategy strategy;

    public int[] sort(int data[]){
        int size = data.length;
        strategy = setSortStrategy(size);
        return strategy.sort(data);
    }

    private SortingStrategy setSortStrategy(int size) {
        if(size > 0 && size < 30){
            System.out.print("Using BubbleSort Strategy: ");
            strategy = new BubbleSort();
        }
        
        if(size >= 30 && size < 100){
            System.out.print("Using InsertionSort Strategy: ");
            strategy = new InsertionSort();
        }
        
        if(size >= 100){
            System.out.print("Using QuickSort Strategy: ");
            strategy = new QuickSort();
        }
        
        return strategy;
    }
}

File: EXAM/bonilla/u3/exam3_final/espe/edu/ec/controller/sorting_context.py
from espe.edu.ec.model.sorting_strategy import BubbleSort, InsertionSort, QuickSort

class SortingContext:
    def __init__(self):
        self.strategy = None

    def set_sort_strategy(self, array_size):
        if 0 < array_size <= 3:
            self.strategy = BubbleSort()
        elif 3 < array_size <= 7:
            self.strategy = InsertionSort()
        else:
            self.strategy = QuickSort()

    def sort(self, data):
        size = len(data)
        self.set_sort_strategy(size)
        return self.strategy.sort(data)
